src/framework/audio/worker/internal/mixer.h

```diff
@@ -93,7 +93,7 @@ class Mixer : public AbstractAudioSource, public Injectable, public async::Async
 
     msecs_t currentTime() const;
 
-    std::unique_ptr<TaskScheduler> m_taskScheduler;
+    std::shared_ptr<TaskScheduler> m_taskScheduler;
```

Comment:
Why shared_ptr? It's not shared with other classes

Reply:
there is a technical problem, unique_ptr does not support the forward declaration, it needs the destructor to be defined.
It looked like there was a forward declaration, but in fact there is not... TaskScheduler is simply included somewhere, maybe because of the unity build.

Reply:
We can convert it to a raw pointer and simply delete it in the destructor

Reply:
Yeah, using a raw pointer might be better